src/fake-timers-src.js

```diff
@@ -1872,6 +1891,228 @@ function withGlobal(_global) {
                 timersModule[nameOfMethodToReplace] =
                     _global[nameOfMethodToReplace];
             }
+            if (clock.timersPromisesModuleMethods !== undefined) {
+                if (nameOfMethodToReplace === "setTimeout") {
+                    clock.timersPromisesModuleMethods.push({
+                        methodName: "setTimeout",
+                        original: timersPromisesModule.setTimeout,
+                    });
+
+                    timersPromisesModule.setTimeout = (
+                        delay,
+                        value,
+                        options = {},
+                    ) =>
+                        new Promise((resolve, reject) => {
+                            const abort = () => {
+                                options.signal.removeEventListener(
+                                    "abort",
+                                    abort,
+                                );
+                                // This is safe, there is no code path that leads to this function
+                                // being invoked before handle has been assigned.
+                                // eslint-disable-next-line no-use-before-define
+                                clock.clearTimeout(handle);
+                                reject(options.signal.reason);
+                            };
+
+                            const handle = clock.setTimeout(() => {
+                                if (options.signal) {
+                                    options.signal.removeEventListener(
+                                        "abort",
+                                        abort,
+                                    );
+                                }
```

Comment:
Could use optional chaining on most of these options.signal blocks to collapse the indentation.
```suggestion
                                  options.signal.?removeEventListener.?(
                                      "abort",
                                      abort,
                                  );
```

Reply:
Sure! I'm all for optional chaining. I'll go through and replace instances of `if (options.signal)` with optional chaining.